tests/unit/geom/PolygonTest.cpp

```diff
@@ -611,4 +611,39 @@ void object::test<42>
     ensure(!poly_->isDimensionStrict(geos::geom::Dimension::L));
 }
 
+
+// test compareToSameClass for polygons including holes
+template<>
+template<>
+void object::test<43>
+()
+{
+    auto geo = reader_.read("POLYGON ((0 0, 10 0, 10 10, 0 10, 0 0), (1 1, 1 2, 2 2, 2 1, 1 1))");
+    ensure(geo != nullptr);
+
+    PolygonPtr poly = dynamic_cast<PolygonPtr>(geo.get());
```

Comment:
You're testing the `Polygon` implementation of `Geometry::compareToSameClass`, so if you want to reduce boilerplate, you can avoid lines like 630-631 (which really test the WKTReader). I would also take out lines like 622 (again, the goal is not to test the `WKTReader`) but it really comes down to preference.

Reply:
Sorry to be dense, but I'm not seeing how to avoid these two lines here?

```
auto geo = reader_.read("POLYGON ((0 0, 10 0, 10 10, 0 10, 0 0), (1 1, 1 2, 2 2, 2 1, 1 1))");
PolygonPtr poly = dynamic_cast<PolygonPtr>(geo.get());
```

(total noob to C++ managed pointers)

Reply:
Well, you can't avoid the first :)

But you can call `geo->compareToSameClass(geo2.get())` directly; you don't need to explicitly cast it to a `Polygon`.

Reply:
Looks like I need to cast the left side, otherwise `compareToSameClass` is inaccessible.  But no need to cast the function parameter, so that simplifies a little.

Reply:
That's interesting -- `compareToSameClass` is `protected` in `Geometry` but `public` in `Polygon`. I can't think of a good reason for that off hand -- I wonder if it's an oversight.

Reply:
Yeah, `Point::compareToSameClass` is also `protected`, so the `Polygon` variant being public is probably a 15-year old oversight. The alternative is to just call `geo->compareTo(geo2.get())` which is really the public interface. Not asking you to change anything, just trying to explain.

Reply:
Thanks @dbaston for helping me understand!  The [comment on it](https://github.com/libgeos/geos/blob/3.9/include/geos/geom/Polygon.h#L137) implies that it was changed intentionally for `Polygon` class?

I think since we're specifically trying to test `compareToSameClass` here, the cast is OK to leave as is.

Reply:
> That's interesting -- `compareToSameClass` is `protected` in `Geometry` but `public` in `Polygon`. I can't think of a good reason for that off hand -- I wonder if it's an oversight.

Sure seems like an oversight/misunderstanding.  Can this be fixed?

Reply:
> Sure seems like an oversight/misunderstanding. Can this be fixed?

Do you mean make it `protected` and then test `compareTo` instead of `compareToSameClass` ?

Reply:
> Do you mean make it `protected` and then test `compareTo` instead of `compareToSameClass` ?

Yes.